src/components/AppNavigationItem/AppNavigationItem.vue

```diff
@@ -123,14 +174,36 @@ export default {
 	},
 	computed: {
 		collapsible() {
-			return this.item.collapsible && this.item.children && this.item.children.length > 0
+			return !this.forceExpand && this.$slots.children
 		},
 		simpleAction() {
 			if (this.collapsible && !this.item.action) {
 				return this.toggleCollapse
 			} else {
 				return this.item.action
 			}
+		},
+		// Checks if the icon prop is an url
+		isIconUrl() {
+			try {
+				return new URL(this.icon)
+			} catch (error) {
+				return false
+			}
+		},
+		// Checks if the component is already a children of another
+		// instance of AppNavigationItem
+		canHaveChildren() {
+			if (this.$parent.$options._componentTag === 'AppNavigationItem') {
+				return false
+			} else return true
+		},
+		// Checks if the component is inside a 'pinned' wrapper
+		// (used to toggle the class 'pinned' in the root element)
+		isPinned() {
+			if (this.$parent.$options._componentTag === 'AppNavigationPinnedItems') {
+				return true
+			} else return false
```

Comment:
Don't mix multiple if-else syntax styles ;)

```suggestion
			} else {
				return false
			}
```

Reply:
done ;)